vignettes/function-flow.Rmd

```diff
@@ -95,63 +95,202 @@ outputs.](images/function-flow-population.png)
+    endocrinological departments (`n_t1d_endo`)
+-   number of type 2 diabetes-specific primary diagnosis codes from
+    endocrinological departments (`n_t2d_endo`)
+-   number of type 1 diabetes-specific primary diagnosis codes from
+    medical departments (`n_t1d_medical`)
+-   number of type 2 diabetes-specific primary diagnosis codes from
+    medical departments (`n_t2d_medical`)
+
+The output is passed to the `get_diagnosis_date()` function for the
+final step of the inclusion process and is subsequently used to classify
+diabetes type.
 
 #### Diabetes-specific podiatrist services
 
 The function `include_podiatrist_services()` uses `sysi` or `sssy` as
 input to extract the dates of all diabetes-specific podiatrist services.
 
-<!-- TODO: Add details on how this filtering should be done -->
+These dates are extracted by filtering values beginning with "54" in the
+`spec` variable of the `sssy` and `sysi` registers by default
+(alternatively, the function can take the `spec2` variable as input
+instead, if that is the data available to the user). In addition,
+services provided to a child of the individual (`barnmak` != 0) are
+excluded using the `barnmak` variable. An internal helper function
+`get_unique_honuge_dates()` is applied to generate a proper date
+variable based on the year-week (wwyy-formatted) variable (`honuge`)
+found in the raw data, and de-duplicates multiple services registered on
+the same date.
+
+`include_podiatrist_services()` outputs a 3-column data frame with one
+row for each individual, containing the following variables:
+
+-   identifier variable (`pnr`)
+-   the date of the first diabetes-specific podiatrist record
+    (`do_podiatrist_1`)
+-   the date of the second diabetes-specific podiatrist record
+    (`do_podiatrist_2`)
+
+The output is passed to the `get_diagnosis_date()` function for the
+final step of the inclusion process.
 
 #### GLD purchases
 
 The function `include_gld_purchases()` uses `lmdb` to extract the dates
-of all GLD purchases (from 1997 onwards).
-
-<!-- TODO: Add details on how this filtering should be done -->
-
-<!-- TODO: Add this + link to resource "For details about this, see [link]." -->
+of all GLD purchases.
+
+These dates are extracted by including all values beginning with "A10" in
+the `atc` variable of the `lmdb` register. Since the diagnosis code data
+on pregnancies (see below) is insufficient to perform censoring prior to
+1997, `include_gld_purchases()` only extracts dates from 1997 onward by
+default (if Medical Birth Register data is available to use for
+censoring, the extraction window can be extended).
+
+This function outputs a `data.frame` with the following variables needed
+later in the classification part of the function flow:
+
+-   identifier variable (`pnr`)
+-   date (`eksd`)
+-   type of drug (`atc`)
+-   amount purchased (`volume` and `apk`)
+-   indication code (`indo`)
+-   brand name or vnr-number (`name` or `vnr`)
```

Comment:
As discussed in other PRs, let's decide on which we want here, rather than complicate it by using an "IF this THEN that" type situation.